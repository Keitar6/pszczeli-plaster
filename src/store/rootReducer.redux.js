import { combineReducers } from "redux";

import { userReducer } from "./userReducer/user.reducer";
import { generalPropReducer } from "./generalPropReducer/generalProp.reducer";
import { cartReducer } from "./cartReducer/cart.reducer";

const rootReducer = combineReducers({
  userReducer,
  generalPropReducer,
  cartReducer,
});
export default rootReducer;
