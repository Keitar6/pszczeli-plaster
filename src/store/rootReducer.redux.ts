import { combineReducers } from "redux";

import { userReducer } from "./userReducer/user.reducer";
import { generalPropReducer } from "./generalPropReducer/generalProp.reducer";
import { cartReducer } from "./cartReducer/cart.reducer";
import { categoriesReducer } from "./categories/category.reducer";
import { orderHistoryReducer } from "./orderHistory/orderHistory.reducer";

export type ReduxState = ReturnType<typeof rootReducer>;

const rootReducer = combineReducers({
  userReducer,
  generalPropReducer,
  cartReducer,
  categoriesReducer,
  orderHistoryReducer
});
export default rootReducer;
