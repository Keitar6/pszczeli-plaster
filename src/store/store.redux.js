import { compose, createStore, applyMiddleware } from "redux";
import logger from "redux-logger";
import rootReducer from "./rootReducer.redux";

const middleWares = [logger];

const composedEnhancer = compose(applyMiddleware(...middleWares));

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
