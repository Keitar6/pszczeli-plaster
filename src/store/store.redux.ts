import {
  compose,
  legacy_createStore as createStore,
  applyMiddleware
} from "redux";
import type { Middleware } from "redux";
import logger from "redux-logger";
import thunk from "redux-thunk";

import rootReducer, { ReduxState } from "./rootReducer.redux";

import { persistStore, persistReducer } from "redux-persist";
import type { PersistConfig } from "redux-persist";
import storage from "redux-persist/lib/storage";

declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__?: typeof compose;
  }
}

type ExtendedPersistConfig = PersistConfig<ReduxState> & {
  whitelist: (keyof ReduxState)[];
};

const persistConfig: ExtendedPersistConfig = {
  key: "root",
  storage: storage,
  whitelist: ["userReducer", "categoriesReducer"],
  blacklist: ["cartReducer"]
};

const middleWares = [
  process.env.NODE_ENV !== "production" && logger,
  thunk
].filter((middleware): middleware is Middleware => Boolean(middleware));

const composeEnhancer =
  (process.env.NODE_ENV !== "production" &&
    window &&
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__) ||
  compose;

const composedEnhancers = composeEnhancer(applyMiddleware(...middleWares));

const persistedReducer = persistReducer(persistConfig, rootReducer);

export const AppStore = createStore(
  persistedReducer,
  undefined,
  composedEnhancers
);

export const persistor = persistStore(AppStore);

export type AppDispatch = typeof AppStore.dispatch;
