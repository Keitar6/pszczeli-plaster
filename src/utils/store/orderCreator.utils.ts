import { DeliveryData, Order, OrderItem } from "store/orderHistory/orderHistory.types";

export const orderCreator= (
  id: string,
  time: string,
  price: number,
  itemsBought: OrderItem[],//{,quantity:15}
  deliveryData: DeliveryData,
  deliveryPrice: number
):Order => {
  return {
    id,
    time,
    price,
    itemsBought,
    deliveryData,
    deliveryPrice
  };
};
