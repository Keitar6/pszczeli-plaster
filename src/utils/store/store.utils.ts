import type { AnyAction } from "redux";

export type ActionWithPayload<T, P> = { type: T; payload: P };

export type Action<T> = { type: T };

export type SomeAction<T, P> = Action<T> | ActionWithPayload<T, P>;

export function actionCreator<T extends string, P>(
  type: T,
  payload: P
): ActionWithPayload<T, P>;

export function actionCreator<T extends string>(
  type: T,
  payload: void
): Action<T>;

export function actionCreator<T extends string, P>(type: T, payload: P) {
  return {
    type,
    payload
  };
}
/*
Funkcja dodająca do funkcji zwracającej actionCreator funkcję sprawdzającą(match(action)) czy 
action otrzymany przez reducer się zgadza
*/
type Matchable<AC extends () => AnyAction & { type: string }> = AC & {
  type: ReturnType<AC>["type"];
  match(action: AnyAction): action is ReturnType<AC>;
};

export function withMatch<
  AC extends (...args: any[]) => AnyAction & { type: string }
>(actionCreator: AC): Matchable<AC>;

export function withMatch<AC extends () => AnyAction & { type: string }>(
  actionCreator: AC
): Matchable<AC>;

export function withMatch<AC extends () => AnyAction & { type: string }>(
  actionCreator: AC
) {
  return Object.assign(actionCreator, {
    match(action: AnyAction) {
      return action.type === actionCreator().type;
    }
  });
}
