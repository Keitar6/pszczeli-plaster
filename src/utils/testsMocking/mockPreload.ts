import { mockCartItems2 } from "./mockCartItem";
import { mockOrderHistory } from "./mockOrderHistory";

export const currentProductCardMock = {
  id: "fsda",
  name: "dobre",
  price: 12,
  image: "dobry",
  dodatki: false,
  quantity: 1
};

export const preload = {
  generalPropReducer: {
    path: "sklep",
    isUserMenuOpened: false,
    isSortingInView:false,
    productCardModal: {
      isProductCardOpened: false,
      currentProductCard: currentProductCardMock
    },
    viewLimiter: 1
  },
  cartReducer: {
    cartItems: [...mockCartItems2],
    isDropdownActive: false,
    _persist: { version: 1, rehydrated: true }
  },
  orderHistoryReducer: {
    delivery: {
      price: 9,
      type: "KurierDHL" as const
    },
    orderHistory: mockOrderHistory,
    isLoading: false,
    error: null
  }
};
