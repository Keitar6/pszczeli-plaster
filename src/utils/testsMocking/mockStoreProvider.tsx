import React, { PropsWithChildren } from "react";
import { render } from "@testing-library/react";
import type { RenderOptions } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import type { PreloadedState } from "@reduxjs/toolkit";

import rootReducer, { ReduxState } from "../../store/rootReducer.redux";
import { Provider } from "react-redux";
import { AppStore } from "../../store/store.redux";
import { BrowserRouter } from "react-router-dom";

// As a basic setup, import your same slice reducers

// This type interface extends the default options for render from RTL, as well
// as allows the user to specify other things such as initialState, store.
interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
  preloadedState?: PreloadedState<ReduxState>;
  store?: typeof AppStore;
}

export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = configureStore({ reducer: rootReducer as any, preloadedState }),
    ...renderOptions
  }: ExtendedRenderOptions = {},
  ifWithRouter = true
) {
  function Wrapper({ children }: PropsWithChildren): JSX.Element {
    if (ifWithRouter)
      return (
        <BrowserRouter>
          <Provider store={store}>{children}</Provider>
        </BrowserRouter>
      );
    else return <Provider store={store}>{children}</Provider>;
  }
  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
