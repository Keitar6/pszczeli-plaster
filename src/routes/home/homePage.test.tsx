import ComponentsRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { AppStore } from "../../store/store.redux";
import HomePage from "./homePage.route";
import { renderWithProviders } from "../../utils/testsMocking/mockStoreProvider";
import { preload } from "../../utils/testsMocking/mockPreload";
import { waitFor } from "@testing-library/react";

const mockDispatch = jest.fn();
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: () => mockDispatch
}));

describe("HomePage", () => {
  const store = AppStore;

  const component = () =>
    renderWithProviders(
      <>
        <HomePage />
      </>,
      {
        preloadedState: {
          ...preload
        }
      }
    );
  beforeEach(() => {
    mockDispatch.mockClear();
  });

  test("useEffect", async () => {
    component();

    await waitFor(() => {
      expect(mockDispatch.mock.calls).not.toEqual([]);
    });
  });

  test("snapshot", () => {
    const homePage = ComponentsRenderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </Provider>
    ).toJSON();
    expect(homePage).toMatchSnapshot();
  });
});
