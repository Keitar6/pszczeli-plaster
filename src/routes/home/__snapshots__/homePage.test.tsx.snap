// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage snapshot 1`] = `
.c4 {
  font-size: 1.95 rem;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  gap: 1rem;
  margin: 2rem 0rem 1rem 0rem;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 2px solid;
  border-image-slice: 1;
  border-image-source: linear-gradient( to left, #023047, #FFB703, #023047 );
  padding-bottom: 0.2rem;
  margin-bottom: 0.5rem;
  position: relative;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  text-align: start;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c3 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 1rem 1rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0rem 0 0 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  />
  <div
    className="c2 c3"
  >
    <h3
      className="c4 c5"
    >
      O NAS
    </h3>
    <div
      className="c6"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio similique laborum excepturi ducimus praesentium, sapiente cumque ipsum perspiciatis quod eos illum aliquam illo a, rem, recusandae veritatis quibusdam in tempora? Lorem ipsum dolor sit amet consectetur adipisicing elit. Optio similique laborum excepturi ducimus praesentium, sapiente cumque ipsum perspiciatis quod eos illum aliquam illo a, rem, recusandae veritatis quibusdam in tempora?
    </div>
  </div>
</div>
`;
